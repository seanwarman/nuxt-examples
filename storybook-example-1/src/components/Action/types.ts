import type { Link } from '@/components/Informational/types'

export type ButtonType = 'primary' | 'secondary' | 'warning' | 'error' | 'success'
export type ButtonSize = 'lg' | 'md' | 'sm' | 'xs'

export interface Button {
    type?: ButtonType
    link?: Link
    theme?: string
    size?: ButtonSize
    iconPosition?: 'before' | 'after'
}