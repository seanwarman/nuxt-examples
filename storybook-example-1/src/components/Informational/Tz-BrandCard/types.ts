type Image = {
  src: string;
  alt?: string;
};

export type BrandCard = {
  title: string;
  image: Image;
  brand: Image;
};
