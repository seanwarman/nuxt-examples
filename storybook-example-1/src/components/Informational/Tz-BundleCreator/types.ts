import type { Media } from '../Tz-ResponsiveImage/types';
type Image = {
  src: string;
  srcSet?: string;
  alt: string;
  width?: number;
  height?: number;
};

type Info = {
  image: Image;
  media: Media[];
  price: string;
  total: number;
  call_to_action: Cta;
};

type Cta = {
  label: string;
};

export type BundleInfo = Info;
