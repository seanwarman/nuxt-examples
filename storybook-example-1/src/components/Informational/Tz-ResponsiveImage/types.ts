import type { Image } from '../types';

export type ImageFormats = {
  webp?: string;
  png?: string;
};

export type Media = {
  width: number | string;
  image: Image;
  src?: string;
  formats: ImageFormats;
  keywords?: string[];
};
