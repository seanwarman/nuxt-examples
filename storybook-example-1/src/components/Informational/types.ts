import type {
  VariantImage,
  Price,
  Layout,
  Reviews
} from '@/components/Informational/Tz-Product-Tile/Tz-Product-Tile.types';

export type Image = {
  width?: string | number;
  height?: string | number;
  src: string;
  alt?: string;
};

export type ImageFormats = {
  webp?: string;
  png?: string;
};

export type Media = {
  width: number | string;
  image: Image;
  src?: string;
  formats: ImageFormats;
  keywords?: string[];
};

export interface LifeStyleImage extends Image {
  lifestyle?: boolean;
}

type FrameName = string;

type Target = '_blank' | '_self' | '_parent' | '_top' | FrameName;

export type Link = {
  url: string | undefined;
  target?: Target;
  internal?: boolean;
};

export interface ProductBasic {
  id: number;
  title: string;
  image: Image;
  uri: string;
  link: never;
}
export interface ProductTile extends ProductBasic {
  price: Price;
  sku: string;
  uri: string;
  stock: number;
  layout?: Layout;
  bike?: boolean;
  lifestyle?: boolean;
  reviews: Reviews;
  brand: never;
}
export interface BrandCard extends ProductBasic {
  brand: Image;
  reviews?: never;
  price: never;
  sku: never;
}
export type Card = ProductTile | BrandCard;
export type Test = BrandCard;

export type DeliverToOption = {
  id: number;
  name: string;
};

export type Discount = {
  name: string;
  amount: number;
  type: DiscountType;
};

export enum DiscountType {
  cycleToWork,
  freeGift,
  vochure
}

export type OrderProduct = {
  line_item_id: number;
  title: string;
  image: Image;
  quantity: number;
  total: number;
};

export type OrderDetails = {
  address: AddressDetails;
  options: DeliveryDetails;
  billing: AddressDetails;
  payment: PaymentMethod;
};

export type AddressDetails = {
  name?: string;
  street_address?: string;
  street_address_2?: string;
  county?: string;
  city?: string;
  postal_code?: string;
};

export type DeliveryDetails = {
  delivery: string;
  description: string;
};

export type PaymentMethod = {
  method: string;
};
