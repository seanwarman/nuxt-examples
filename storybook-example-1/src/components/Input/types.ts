export type Option = {
  name?: string;
  value?: string | number | boolean;
  prefix?: string;
};

export interface TzSelectProps {
  description?: string;
  options: Option[];
  defaultState?: Option;
  size?: string;
  prefixes?: boolean;
}
