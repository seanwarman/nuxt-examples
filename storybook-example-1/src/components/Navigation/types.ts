type Image = {
  src: string;
  alt: string;
  width?: number;
  height?: number;
};

type Icon = {
  text: string;
};

type Button = {
  label: string;
  link: string;
};

type Tab = {
  index: number;
  name: string;
};

type NavItemContent = {
  image: Image;
  button: Button;
};

type ItemHeader = {
  image: Image;
};

type Active = {
  index: number;
  link: string;
};

type NavigationMenu = {
  name: string;
  links: NavItem[];
  header?: NavItemHeader;
};

type Underline = {
  width: number | string;
  height: number | string;
  left: string;
  top: number | string;
};

enum Direction {
  Up = 'up',
  Down = 'down',
  Left = 'left'
}

export interface Item {
  image?: Image;
  icon?: Icon;
  label: string;
  link: string;
  menu: NavItem[];
  list?: NavItem[];
  content?: NavItemContent;
  active?: boolean;
  header?: ItemHeader;
}

export type NavImage = Image;
export type NavIcon = Icon;
export type NavTab = Tab;
export type NavItem = Item;
export type NavContent = NavItemContent;
export type NavItemHeader = ItemHeader;
export type NavMenu = NavigationMenu;
export type NavUnderline = Underline;
