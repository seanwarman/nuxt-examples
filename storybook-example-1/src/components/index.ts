export { default as TzButton } from './Action/Tz-Button/Tz-Button.vue';
export { default as TzCheckoutCta } from './Action/Tz-Checkout-CTA/Tz-Checkout-Cta.vue';
export { default as TzModalDialog } from './Action/Tz-ModalDialog/Tz-ModalDialog.vue';
export { default as TzPayPalButton } from './Action/Tz-Button/Tz-PayPalButton.vue';
export { default as TzProgressBar } from './Action/Tz-Progress-Bar/Tz-Progress-Bar.vue';

export { default as TzBasketAccordion } from './Informational/Tz-Basket-Accordion/Tz-Basket-Accordion.vue';
export { default as TzBasketCard } from './Informational/Tz-Basket-Card/Tz-Basket-Card.vue';
export { default as TzBasketTotalsSummary } from './Informational/Tz-Basket-Totals-Summary/Tz-Basket-Totals-Summary.vue';
export { default as TzBundleCreator } from './Informational/Tz-BundleCreator/Tz-BundleCreator.vue';
export { default as TzC2wBanner } from './Informational/Tz-C2W-Banner/Tz-C2W-Banner.vue';
export { default as TzExpressCheckout } from './Informational/Tz-ExpressCheckout/Tz-ExpressCheckout.vue';
export { default as TzInfoBox } from './Informational/Tz-InfoBox/Tz-InfoBox.vue';
export { default as TzJumboCard } from './Informational/Tz-Jumbo-Card/Tz-Jumbo-Card.vue';
export { default as TzOrderItemsSummary } from './Informational/Tz-Order-Items-Summary/Tz-Order-Items-Summary.vue';
export { default as TzOrderTotalsSummary } from './Informational/Tz-Order-Totals-Summary/Tz-Order-Totals-Summary.vue';
export { default as TzProductTile } from './Informational/Tz-Product-Tile/Tz-Product-Tile.vue';
export { default as TzUspRibbon } from './Informational/Tz-USP-Ribbon/Tz-USP-Ribbon.vue';

export { default as TzInputField } from './Input/Tz-InputField/Tz-InputField.vue';
export { default as TzLogin } from './Input/Form/Tz-Login/Tz-Login.vue';
export { default as TzPaymentMethods } from './Input/Tz-PaymentMethods/Tz-PaymentMethods.vue';
export { default as TzRadioField } from './Input/Tz-RadioField/Tz-RadioField.vue';
export { default as TzSelectField } from './Input/Tz-SelectField/Tz-SelectField.vue';

export { default as TzNavigation } from './Navigation/Tz-Navigation/Tz-Navigation.vue';

export { default as TzContainer } from './Structural/Tz-Container/Tz-Container.vue';

export { default as TzAccountIcon } from './Informational/Icons/Tz-Account-Icon.vue';
export { default as TzArrowIcon } from './Informational/Icons/Tz-Arrow-Icon.vue';
export { default as TzBasketIcon } from './Informational/Icons/Tz-Basket-Icon.vue';
export { default as TzCaretIcon } from './Informational/Icons/Tz-Caret-Icon.vue';
export { default as TzCheckmarkIcon } from './Informational/Icons/Tz-Checkmark-Icon.vue';
export { default as TzChevronIcon } from './Informational/Icons/Tz-Chevron-Icon.vue';
export { default as TzCloseIcon } from './Informational/Icons/Tz-Close-Icon.vue';
export { default as TzExclaimationIcon } from './Informational/Icons/Tz-Exclaimation-Icon.vue';
export { default as TzExpandIcon } from './Informational/Icons/Tz-Expand-Icon.vue';
export { default as TzFilterIcon } from './Informational/Icons/Tz-Filter-Icon.vue';
export { default as TzSearchIcon } from './Informational/Icons/Tz-Search-Icon.vue';
export { default as TzWishlistIcon } from './Informational/Icons/Tz-Wishlist-Icon.vue';
