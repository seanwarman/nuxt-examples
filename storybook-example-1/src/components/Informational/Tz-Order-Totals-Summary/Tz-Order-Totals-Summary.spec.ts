/**
 * @vitest-environment jsdom
 */

import { mount, VueWrapper } from '@vue/test-utils';
import { expect, describe, beforeEach, it } from 'vitest';
import { TzOrderTotalsSummary } from '@/components';

import { DiscountType } from '../types';

let wrapper: VueWrapper;

describe('TzInfoBox', () => {
  beforeEach(() => {
    wrapper = mount(TzOrderTotalsSummary, {
      props: {
        noOfItems: 1,
        discounts: [
          {
            name: '',
            amount: 29,
            type: DiscountType.cycleToWork
          }
        ],
        savings: 0,
        subTotal: 100,
        shippingCost: 0,
        totel: 100
      }
    });
  });

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
