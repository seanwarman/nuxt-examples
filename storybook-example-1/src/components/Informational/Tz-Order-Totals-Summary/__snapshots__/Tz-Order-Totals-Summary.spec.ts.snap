// Vitest Snapshot v1

exports[`TzInfoBox > renders correctly 1`] = `
<section
  class="flex flex-col gap-0.5 text-sm font-normal uppercase text-gray-500"
  totel="100"
>
  <dl
    class="flex flex-col items-stretch"
  >
    <div
      class="flex items-center justify-between"
    >
      <dt>
        Sub total (1 item)
      </dt>
      <dd
        class="text-right"
      >
        £100.00
      </dd>
    </div>
  </dl>
  <dl
    class="flex flex-col items-stretch gap-0.5"
  >
    
    <div
      class="flex items-center justify-between text-primary"
    >
      <dt />
      <dd
        class="text-right"
      >
        - £29.00
      </dd>
    </div>
    
  </dl>
  <section>
    <dl
      class="mt-2 flex flex-col items-stretch"
    >
      <div
        class="flex items-center justify-between"
      >
        <dt>
          Delivery
        </dt>
        <dd
          class="text-right"
        >
          £0.00
        </dd>
      </div>
    </dl>
  </section>
  <dl
    class="mt-4 flex flex-col items-stretch"
  >
    <div
      class="flex items-center justify-between text-gray-800"
    >
      <dt
        class="text-lg font-extrabold"
      >
        Total
      </dt>
      <dd
        class="text-lg font-extrabold"
      >
        £NaN
      </dd>
    </div>
  </dl>
</section>
`;
